Dockerfile

```diff
@@ -0,0 +1,6 @@
+FROM ubuntu:xenial
+
+RUN apt-get update
+RUN apt-get install -y git cmake ctags tmux
+RUN apt-get install -y libusb-1.0-0-dev binutils-arm-none-eabi gcc-arm-none-eabi libnewlib-arm-none-eabi dfu-util
+RUN apt-get install -y python3 python3-pil
```

Comment:
This looks great!

You might want to try out the clang build as well (travis is broken so I haven't been able to add it). But I won't hold up this pull request for that.

An FYI, I have a large change-set in the ktype branch that includes a bunch of build related improvements. 

Mainly this includes a bootloader space check. I'm pretty sure this will fail with Ubuntu's version of gcc so we might have to move to a distro with a more modern gcc as the example (Arch Linux for example).
Everything else will build just fine, so I'm not terribly worried at the moment.